### PR TITLE
Upgrade to Closure Compile 2015-05-05

### PR DESCRIPTION
**Pull request by vandenoever**

### CODE REVIEW

**Comment by thz** [comment]
retest this please

**Comment by vandenoever** [comment]
```
[ 71%] Generating benchmark.js-target
ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$AlignCurrentSelectionJustified" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$Benchmark" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$BenchmarkAction" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$BoldCurrentSelection" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$EnterEditMode" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$HTMLActionResult" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$HTMLResultsRenderer" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$InsertLetterA" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$MoveCursorLeft" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$MoveCursorToEnd" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$MoveCursorToEndDirect" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$MoveCursorToStart" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$OdfBenchmarkContext" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$OpenDocument" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$PreloadDocument" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$RangeCalculator" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$RemoveCurrentSelection" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$RemovePositions" never provided

ERROR - required entry point "module$$home$jenkins$workspace$WebODF_PullReq$programs$benchmark$SelectEntireDocument" never provided
19 error(s), 0 warning(s)
```
This problem first showed up in closure compiler version 20140303.

File: webodf/lib/gui/DirectFormattingController.js [hunk 1]
```diff
@@ -856,6 +856,13 @@ gui.DirectFormattingController = function DirectFormattingController(
     /**
      * @return {!boolean}
      */
+    this.emptyBoolFunction = function () {
```
Comment (peitschie) [inline webodf/lib/gui/DirectFormattingController.js]: Is there a reason this function needs to be exposed on ``this``?

**Comment by peitschie** [comment]
Apart from that one tiny point, the patch looks good to me. I believe this has been brewing for, what, 1.5 years? Either way, it's matured nicely... thanks for wrestling with the benchmark builds as well @vandenoever !

One small fix up, then this shall be ready to sail into master in my opinoni.

**Comment by vandenoever** [comment]
@peitschie the this is not needed indeed. CC made me think it by writing 'this' as a desired attribute for the function signature in the error message. So I've removed it now.
One other thing that could be improved is to have a patch for the externs instead of copying them over. So unpack them from compilar.jar/externs.zip and then adapt the parts that we disagree on.
But that can go in a subsequent pr.

**Comment by vandenoever** [comment]
So after 1.5 years, we have a new CC.
No real new errors in the code were discovered by this new version.

**Comment by peitschie** [comment]
:balloon:  :balloon:  :balloon:  :tada: :birthday: :balloon:

**Pull request merged**